protocol/src/main/java/org/opensearch/sql/protocol/response/format/FlatResponseFormatter.java

```diff
@@ -55,22 +58,54 @@ public String format(Throwable t) {
   static class FlatResult {
     private final QueryResult response;
     private final boolean sanitize;
+    private final boolean pretty;
+
+    private List<String> headers;
+    private List<List<String>> data;
+    private int[] maxWidths;
```

Comment:
Could you refactor this class a little? I think this class was focus on sanitize only. After this, the class becomes more and more complex...

Reply:
Although this is existing code, I don't think having inner static classes inside is a good pattern. While refactoring can we separte it out and have individual classes for Sanitizer and Prettifier if possible[Up to you how you refactor].

Reply:
Could you add Javadoc for new class? In future, I think we can refactor base, sanitizer, prettier by decorator as needed, e.g. sanitize and then pretty.